### PR TITLE
Fixes #12102 - Adds socket errors catching in rackspace test connection function

### PR DESCRIPTION
**Pull request by amirfefer**

### CODE REVIEW

**Comment by amirfefer** [comment]
By the way, do you think it necessary to add foreman error code ?

**Comment by dLobatog** [comment]
@amirfefer It's fine, `e.message` should contain the actual error. 

What are these `StandardError` rescues for exactly? 
Could you write a comment on the method explaining how did you get to them?  If the comment would be very long, a proper description in the commit message so anyone can see it through `git blame` would work too.
Can you rescue more specific exceptions?

Thanks!

**Comment by amirfefer** [comment]
@eLobato I change it to a more specific exception `<Excon::Errors::SocketError>` which catch connection problems, such as invalid server address.

File: app/models/compute_resources/foreman/model/ec2.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@ def zones
     def test_connection(options = {})
       super
       errors[:user].empty? and errors[:password].empty? and regions
-    rescue Fog::Compute::AWS::Error => e
```
Comment (dLobatog) [inline app/models/compute_resources/foreman/model/ec2.rb:74]: No need to remove the `=> e` here, especially if it's used below on `e.message`

**Comment by dLobatog** [comment]
@amirfefer Thanks, there's a tiny issue on ec2.rb I think. Aside from that, do you mind to write a description of what you're doing in your commit? Good to merge after that :smile:

**Comment by dLobatog** [comment]
Merged as 7a125e2fb9e6dd76039fa1ee9bed365e9db79a0b, thanks @amirfefer!

**Pull request closed without merging**